docs/contributing/contribution_checklist.rst

```diff
@@ -62,7 +62,11 @@ When you’re done with your work...
 
     * Push to the remote fork of your repo.
     * Follow `these instructions <https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork>`__ to create a PR from your commit.
-    *  In the PR, choose a short title which sums up the changes that you have made, and in the body provide more details about what your changes do. Also mention the number of the issue where discussion has taken place, e.g. "Closes #123".
+    * Provide background for PR reviewers:
```

Comment:
This text is awesome.

I'd suggest a small change in motivation:
"Provide background for reviewers so they can understand and approve your PR more quickly:"

Please break out the "Add the tag" stuff in its own bullet. Sub-bullets for the tags might make it more scannable, too.